core/src/main/java/io/aiven/klaw/service/OperationalRequestsService.java

```diff
@@ -0,0 +1,337 @@
+      int startVar = (requestPageNo - 1) * recsPerPage;
+      int lastVar = (requestPageNo) * (recsPerPage);
+
+      List<String> numList = new ArrayList<>();
+      commonUtilsService.getAllPagesList(pageNo, currentPage, totalPages, numList);
+
+      for (int i = Math.max(0, startVar); i < Math.min(lastVar, totalRecs); i++) {
+        OperationalRequest operationalRequest = origActivityList.get(i);
+        operationalRequest.setAllPageNos(numList);
+        operationalRequest.setTotalNoPages("" + totalPages);
+        operationalRequest.setCurrentPage(pageNo);
+        envSelected = getEnvDetails(operationalRequest.getEnvironment());
+        operationalRequest.setEnvironmentName(envSelected.getName());
+        newList.add(operationalRequest);
+      }
+    }
+
+    return newList;
+  }
+
+  public Env getEnvDetails(String envId) {
+    Optional<Env> envFound =
+        manageDatabase.getKafkaEnvList(commonUtilsService.getTenantId(getUserName())).stream()
+            .filter(env -> Objects.equals(env.getId(), envId))
+            .findFirst();
+    return envFound.orElse(null);
+  }
+
+  private List<OperationalRequest> filterByTenantAndSort(
+      Order order, String userName, List<OperationalRequest> topicReqs) {
+    try {
+      final Set<String> allowedEnvIdSet = commonUtilsService.getEnvsFromUserId(userName);
+      topicReqs =
+          topicReqs.stream()
+              .filter(topicRequest -> allowedEnvIdSet.contains(topicRequest.getEnvironment()))
+              .sorted(getPreferredOrdering(order))
+              .collect(Collectors.toList());
+    } catch (Exception e) {
+      log.error("No environments/clusters found.", e);
+      return new ArrayList<>();
+    }
+    return topicReqs;
+  }
+
+  private Comparator<OperationalRequest> getPreferredOrdering(Order order) {
+    return switch (order) {
+      case ASC_REQUESTED_TIME -> compareByTime();
+      case DESC_REQUESTED_TIME -> Collections.reverseOrder(compareByTime());
+    };
+  }
+
+  private static Comparator<OperationalRequest> compareByTime() {
+    return Comparator.comparing(OperationalRequest::getRequesttime);
+  }
+
+  public ApiResponse resetConsumerOffsets(String req_no) {
+    log.info("approveConsumerOffsetRequests {}", req_no);
+    final String userDetails = getUserName();
+    int tenantId = commonUtilsService.getTenantId(userDetails);
+    if (commonUtilsService.isNotAuthorizedUser(
+        getPrincipal(), PermissionType.APPROVE_SUBSCRIPTIONS)) {
+      return ApiResponse.NOT_AUTHORIZED;
+    }
+    ResetConsumerGroupOffsetsRequest resetConsumerGroupOffsetsRequest =
+        ResetConsumerGroupOffsetsRequest.builder().build();
+
+    HandleDbRequests dbHandle = manageDatabase.getHandleDbRequests();
+    OperationalRequest operationalRequest =
+        dbHandle.getOperationalRequest(Integer.parseInt(req_no), tenantId);
+    ApiResponse apiResponse;
+    try {
+      apiResponse =
+          clusterApiService.resetConsumerOffsets(
```

Comment:
Could we make the response a POJO so we know it should be returned as a particular object? might make it easier to use.

Reply:
Initially I had a pojo, and after that being written into ApiResponse object, from cluster api to core, the object lost it's semantics probably. so had to remove the pojo. 

I still have to make more changes on this feature, so will try to get the pojo back.